### ISSUE TITLE
Nested `pointer-event` rules are not respected

### ISSUE THREAD

**Issue by tortilaman** [capricorn86/happy-dom#956]
**Describe the bug**
The Dialog component from `@radix-ui/react-dialog` adds a `pointer-events: none` style rule to the body tag, and a `pointer-events: auto` style rule to the dialog's content element, which is a child of the body. When running `@testing-library/react` with `happy-dom`, the following error is thrown:

<img width="743" alt="image" src="https://github.com/capricorn86/happy-dom/assets/5018268/ec9e3f23-16cd-4f47-bd5f-dda430abf3ff">

When I switched my setup from `happy-dom` to `jsdom` for this specific test, the test passed.

**To Reproduce**

**Expected behavior**
The test should be able to reach the element which is a child of the container with `pointer-events: auto`.

**Screenshots**

**Device:**
 - OS: MacOS Ventura
 - Browser: Vitest with happy-dom
 - Version: Vitest v0.30.1, happy-dom v9.0.0

**Additional context**

**Comment by capricorn86** [comment]
Thanks for reporting @tortilaman! :slightly_smiling_face: 

I believe that the problem is that JSDOM didn't have support for inherited CSS properties before. They just released basic support in v21.1.1.

For me it worked when I opened the dialog before attempting to access it's content using testing library, as Radix UI will then set "pointer-events: auto".

**Example:**
```javascript
    <Dialog.Root open={true}>
          ....
    </Dialog.Root>
```

**Comment by capricorn86** [comment]
I will close this as it doesn't seem to be a bug. Please re-open if the answer didn't resolve it for you.
